roadmap/implementers-guide/src/types/candidate.md

```diff
@@ -7,6 +7,14 @@ In a way, this entire guide is about these candidates: how they are scheduled, c
 
 This section will describe the base candidate type, its components, and variants that contain extra data.
 
+## Para Id
+
+A unique 32-bit identifier referring to a specific para (chain or thread). The relay-chain runtime guarantees that `ParaId`s are unique for the duration of any session, but recycling and reuse over a longer period of time is permitted.
```

Comment:
It doesn't matter for the purpose of this PR, but I'm curious: are the wrapped `u32`s hashes of some kind, or handed out sequentially, or what?

Reply:
We haven't described the registrar in the guide yet, but I think it gives them out sequentially. Parachain IDs start at `0` and parathread IDs start with some of the higher bits set, although I don't remember exactly what.

Reuse is fine as long as it's at least a few sessions apart, although it would be best not to reuse until having cycled completely. We could alternatively use a generation/index system like this for handing out IDs to avoid reuse over a long period of time: http://bitsquid.blogspot.com/2014/08/building-data-oriented-entity-system.html

The uniqueness property is what's important here and the details of the registrar are free to change